### PR TITLE
Add metric_cost in config.

### PR DESCRIPTION
**Pull request by chowchow316**
First try, not sure how to use selector, and will add QuotaRule later.

### CODE REVIEW

File: contrib/endpoints/src/api_manager/config.cc [hunk 1]
```diff
@@ -398,6 +398,14 @@ bool Config::LoadService(ApiManagerEnvInterface *env,
     string tf;
     ::google::protobuf::TextFormat::PrintToString(service_, &tf);
     env->LogDebug(tf.c_str());
+
+    auto quota_info = service_.quota();
+    for (auto &metric_rule : quota_info.metric_rules()) {
+      for (auto &metric_cost : metric_rule.metric_costs()) {
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/config.cc]: not here. you need to create a separate function like

bool Config::LoadQuotaRule(ApiManagerEnvInterface *env) {
  for (auto &rule : service_.quota().metric_rules()) {
    auto method = utils::FindOrNull(method_map_, rule.selector());
    if (method) {
             for (auto &metric_cost : rule.metric_costs()) {
                    method->add_quota_metric(metrick_cost.first, metric_cost.second;
            }
    } else {
       log error
    }
  }
  

    for (auto &rule : service_.quota().quota_rules()) {
          auto method = utils::FindOrNull(method_map_, rule.selector());
           if (method) {
               for (const auto& rule: rule.groups) {
                         // find out the group that rule.group point
                         for (const auto& limit : group.limits()) {
                              method->add_metric(limit.metric, rule.cost);

File: contrib/endpoints/src/api_manager/config.h [hunk 2]
```diff
@@ -129,6 +129,8 @@ class Config {
   // jwksUri for the issuer. It is set to true if jwksUri is not provided in
   // service config and we have not tried openId discovery to fetch jwksUri.
   std::map<std::string, std::pair<std::string, bool>> issuer_jwks_uri_map_;
+
+  std::map<std::string, int64_t> metric_cost_;
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/config.h]: no here

File: contrib/endpoints/src/api_manager/method_impl.h [hunk 3]
```diff
@@ -62,6 +62,10 @@ class MethodInfoImpl : public MethodInfo {
 
   const std::string &backend_address() const { return backend_address_; }
 
+  const std::map<std::string, int64_t> && metric_cost() {
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/method_impl.h]: const std::map<std::string, int64_t>& metric_cost_map() const { return metric_cost_map_;  }

File: contrib/endpoints/src/api_manager/method_impl.h [hunk 4]
```diff
@@ -96,6 +100,10 @@ class MethodInfoImpl : public MethodInfo {
 
   void set_selector(const std::string &selector) { selector_ = selector; }
 
+  void set_metric_cost(const std::map<std::string, int64_t> &&metric_cost) {
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/method_impl.h]: void add_metric_cost(string metric_name, int64 cost) { }

**Comment by chowchow316** [comment]
Duplicated.

**Pull request closed without merging**